PROShine/Views/BattleView.xaml.cs

```diff
@@ -0,0 +1,410 @@
+        private double _opponentHPWidth;
+        private double _activeHPWidth;
+        private double _expBarWidth;
+
+        public BattleView(BotClient bot, MainWindow parent)
+        {
+            _bot = bot;
+            _parent = parent;
+            DataContext = this;
+            InitializeComponent();
+
+            _hpGradient = new GradientStopCollection
+            {
+                new GradientStop(Colors.Red, 0),
+                new GradientStop(Colors.Red, 0.15),
+                new GradientStop(Colors.Yellow, 0.3),
+                new GradientStop(Colors.Yellow, 0.5),
+                new GradientStop(Colors.Green, 0.6),
+                new GradientStop(Colors.Green, 1)
+            };
+        }
+
+        public void OnBattleUpdate()
+        {
+            Dispatcher.InvokeAsync(delegate
+            {
+                if (_bot.Game.ActiveBattle == null)
+                    return;
+                OpponentName.Text = PokemonNamesManager.Instance.Names[_bot.Game.ActiveBattle.OpponentId];
+                if (_bot.Game.ActiveBattle.IsShiny)
+                    OpponentName.Text = "Shiny " + OpponentName.Text;
+                if (_bot.Game.ActiveBattle.IsWild)
+                    OpponentName.Text = "Wild " + OpponentName.Text;
+                OpponentCaughtIcon.Visibility = _bot.Game.ActiveBattle.AlreadyCaught ? Visibility.Visible : Visibility.Hidden;
+                OpponentLevel.Text = _bot.Game.ActiveBattle.OpponentLevel.ToString();
+                OpponentMaxHealth.Text = _bot.Game.ActiveBattle.OpponentHealth.ToString();
+                OpponentCurrentHealth.Text = _bot.Game.ActiveBattle.CurrentHealth.ToString();
+
+                if (_bot.Game.ActiveBattle.OpponentStatus.ToLowerInvariant() == "none")
+                    OpponentStatus.Text = "";
+                else
+                    OpponentStatus.Text = _bot.Game.ActiveBattle.OpponentStatus;
+
+                OpponentType1.Text = TypesManager.Instance.Type1[_bot.Game.ActiveBattle.OpponentId].ToString();
+                OpponentType2.Text = TypesManager.Instance.Type2[_bot.Game.ActiveBattle.OpponentId].ToString();
+
+                string gender = _bot.Game.ActiveBattle.OpponentGender;
+                OpponentGender.Icon = gender == "M" ? FontAwesomeIcon.Mars : gender == "F" ? FontAwesomeIcon.Venus : FontAwesomeIcon.Question;
+                OpponentForm.Text = _bot.Game.ActiveBattle.AlternateForm.ToString();
+
+                PokemonStats stats = EffortValuesManager.Instance.BattleValues[_bot.Game.ActiveBattle.OpponentId];
+                List<string> evs = new List<string>();
+
+                foreach (StatType type in Enum.GetValues(typeof(StatType)).Cast<StatType>())
+                {
+                    int ev = stats.GetStat(type);
+                    if (ev > 0)
+                        evs.Add($"{type}: {ev}");
+                }
+
+                OpponentEVs.ToolTip = string.Join("&#xD;", evs);
```

Comment:
Is this working by any chance? It's a bit weird to have an hexadecimal constant in the code.
```
string.Join(Environment.NewLine, evs);
```

Reply:
Sure enough, thanks